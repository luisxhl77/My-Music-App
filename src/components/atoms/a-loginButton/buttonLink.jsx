import PropTypes from 'prop-types';
import './buttonLink.scss';

export const ButtonLink = ({name, link}) => {
  return (
    <a aria-label='buttonLink' href={link} className='buttonLink'>{name}</a>
  )
}

ButtonLink.propTypes = {
  name: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
}