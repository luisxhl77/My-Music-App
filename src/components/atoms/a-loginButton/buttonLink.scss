@import '../../../styles/variables.scss';
.buttonLink{
    width: 140px;
    height: 15px;
    padding: 20px;
    border-radius: 40px;
    color: $text-one;
    text-align: center;
    text-transform: uppercase;
    background-color: #1db932;
    &:hover{
        box-shadow: 2px 2px 20px #00ff22;
    }
    @media (min-width: $medium) {
        width: 170px;
        height: 20px;
    }
}