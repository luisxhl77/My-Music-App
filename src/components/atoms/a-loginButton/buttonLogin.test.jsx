import { render, screen } from "@testing-library/react"
import { ButtonLink } from "./buttonLink";

describe('Pruebas de <ButtonLink/>', () => {

    const name = 'iniciar sesion'; 
    const link = 'https://login.com/';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <ButtonLink name={name} link={link}/>)
        expect(container).toMatchSnapshot();

    })

    test('debe de mostrar el nombre del boton', () => {
        
        render( <ButtonLink name={name} link={link}/>)
        
        expect( screen.getByText(name) ).toBe(screen.getByLabelText('buttonLink'));

    })
    
    test('debe de mostrar el URL del boton', () => {
        
        render( <ButtonLink name={name} link={link}/>)
        
        const { href } = screen.getByLabelText('buttonLink');
        expect( href ).toBe( link );

    })

})