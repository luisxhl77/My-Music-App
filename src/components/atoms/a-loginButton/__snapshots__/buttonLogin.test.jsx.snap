// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <ButtonLink/> debe de hacer match con el snapshot 1`] = `
<div>
  <a
    aria-label="buttonLink"
    class="buttonLink"
    href="https://login.com/"
  >
    iniciar sesion
  </a>
</div>
`;
